src/magic_folder/config.py

```diff
@@ -219,14 +243,76 @@ def create_global_configuration(basedir, api_endpoint_str, tahoe_node_directory,
         )
 
     config = GlobalConfigDatabase(
+        basedir=basedir,
         database=connection,
-        api_token_path=basedir.child(b"api_token"),
+        token_provider=FilesystemTokenProvider(
+            basedir.child(b"api_token"),
+        )
     )
     # make sure we have an API token
     config.rotate_api_token()
     return config
 
 
+def create_testing_configuration(basedir, tahoe_node_directory, global_service):
```

Comment:
I think the need for this function goes away if `magic_folder_list` and others accept an http client to use as an argument.  The tests can then use a regular global config and `treq_for_folders` or similar to make an http client to supply.